composeApp/build.gradle.kts

```diff
@@ -71,16 +99,24 @@ kotlin {
     }
 }
 
+compose.resources {
+    customDirectory(
+        sourceSetName = "commonMain",
+        directoryProvider = provider { layout.projectDirectory.dir(config.resRoot) },
+    )
+}
+
 android {
     namespace = "org.ooni.probe"
     compileSdk = libs.versions.android.compileSdk.get().toInt()
 
     defaultConfig {
-        applicationId = "org.ooni.probe"
+        applicationId = config.appId
         minSdk = libs.versions.android.minSdk.get().toInt()
         targetSdk = libs.versions.android.targetSdk.get().toInt()
         versionCode = 1
         versionName = "1.0"
+        resValue("string", "app_name", config.appName)
```

Comment:
This code is only running for Android isn't it? Plus you are already doing this through the `strings.xml` file.

What we're not covering is having the app name with "Debug" for our debug builds.

The `applicationIdSuffix = ".debug"` config is also only for Android. We might need something similar for iOS?

Reply:
I removed the `app_name` entry from `composeApp/src/androidMain/res/values/strings.xml` which was setting it statically.

Compose strings set in `composeApp/src/dwMain/resources/values/strings.xml` and  `composeApp/src/ooniMain/resources/values/strings.xml` does not influence the manifest file. So i resulted to setting the value here.

Android configuration is by code. For iOS, I have to do custom scheme configuration in Xcode which is not the prettiest thing to do. So i prefer to do it last. We can definitely have a separate debug bundle ID for iOS.